packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -77,12 +87,19 @@ export default class ConfigParser {
              */
             this._config = merge(detectBackend(this._config), this._config, MERGE_OPTIONS)
 
+            /**
+             * compile files if Babel or TypeScript are installed
+             */
+            if (!loadTypeScriptCompiler(this._config.tsNodeOpts) && !loadBabelCompiler()) {
```

Comment:
This should happen before we call `addConfigEntry` otherwise any imports or TS code in the config can't be compiled, no?

Reply:
Great question! I moved it here so that we could have the options parsed to hand to ts-node. Is there a better spot off-hand we could put it, or do we need to minimally open the config to get the options and then re-parse after?

Reply:
Ah yes, we can't read the `tsNodeOptions` before we actually have compiled it for which we need the `tsNodeOptions`. A typical chicken&egg problem 😄 It seems to be that the compiler options can be passed in as environment variables. I would suggest to keep things as they are and instead add some documentation as to how to modify these options.

We have received a lot of comments about how user can use imports in their config file and this solution should make this very easy. So I would love to be able to use TypeScript in the config without having to set up a special way to call wdio.

Reply:
Oooh, using environment variables is a great idea! Confirmed it works on our end too. 🎉 

OK I want to make sure I understand the new direction, are we saying:

1) Remove the new tsNodeOpts config stuff and add docs instructing to use environment variables for any specific config? https://github.com/TypeStrong/ts-node#cli-and-programmatic-options seems to give one for each so this should be sufficient.

Or are we saying:

2) Keep the new tsNodeOpts config stuff and also add docs letting people know about both options.

Reply:
Environment variables can be pain in daily tests development routines. I don't like the idea when I need set env variable before I write\debug new test when I run `yarn run wdio --spec` every day.
It could sound crazy but I would like to have another config file e.g. `.wdiocompile` to control wdio autocompile behavior

Reply:
Thanks for joining in! :)

.wdiocompile might be a good idea to also solve the chicken-and-egg config parsing problem too.

Another option would be to nested tsNodeOpts under autoCompileOpts, but I like how a dedicated config file would solve the parsing issue of the rest of the config, just wanted to mention it as an option though

Reply:
I think we should advocate both options. The important piece is that people can write TS in their config file and have the ability to add type safety to it. So maybe the tsconfig options are not really necessary for just compiling the config but when running the tests. To give more context: the config is read 2x:

1. in the launcher (main process) to read services and run the onPrepare and onComplete hooks
2. within the worker process to establish the framework, reporters and everything else

@r4j4h if you could find out whether you need the ts-node options also for just compiling the config, that would be great. Otherwise we have to be clear in the docs that only environment variables can be used if you have the need to set these configs.

Reply:
@christian-bromann Great distinction. I would hazard a guess that it is only when running the tests, as it is after chromedriver is initialized:

```
2021-02-01T18:52:17.624Z DEBUG @wdio/config:utils: Found 'ts-node' package, auto-compiling TypeScript files

Execution of 1 spec files started at 2021-02-01T18:52:17.831Z

Starting ChromeDriver 87.0.4280.88 (89e2380a3e36c3464b5dd1302349b1382549290d-refs/branch-heads/4280@{#1761}) on port 9515
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
[1612205537.997][SEVERE]: bind() failed: Cannot assign requested address (99)
[0-0] RUNNING in chrome - /features/testing.feature
```

I am making pretty good progress toward .wdiocompile as well, for what that is worth

Reply:
Great! So we can keep the `tsNodeOpts` in our WebdriverIO Testrunner config and add to the docs that if people are interested adding specific compile options for the config they can through environment variables.